tests/cuda_ndarrays/conftest.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=arguments-differ, inconsistent-return-statements, protected-access, abstract-method/
+import subprocess
+import os
+import pathlib
+import sys
+import shutil
+import pytest
+
+NEEDS_FROM_PARENT = hasattr(pytest.Item, "from_parent")
+
+def pytest_collect_file(parent, path):
+    """
+    A hook to collect test_*.cu test files.
+
+    """
+    if path.ext == ".cu" and path.basename.startswith("test_"):
+        if NEEDS_FROM_PARENT:
+            return CTestFile.from_parent(path=pathlib.Path(path), parent=parent)
+        return CTestFile(parent=parent, path=pathlib.Path(path))
+
+def pytest_collection_modifyitems(items):
+    """
+    a hook to modify the items before the tests for C Cuda unit test files.
+
+    """
+    for item in items:
+        if item.fspath.ext == ".cu":
+            item._nodeid = item.nodeid + " < " + item.test_result["DSCR"] + " >"
+
+class CTestFile(pytest.File):
+    """
+    A custom file handler class for C unit test files.
+
+    """
+
+    @classmethod
+    def from_parent(cls, **kwargs):
+        return super().from_parent(**kwargs)
+
+    def collect(self):
+        """
+        Overridden collect method to collect the results from each
+        C unit test executable.
+

```

Comment:
Unnecessary empty line in docstring

Reply:
Fixed 2a376d0a156d1ca4ca39fdd6c244c4b0b145c000